### ISSUE TITLE
[RFC] (UN)STABLE flag keywords

### ISSUE THREAD

**Issue by vladfaust** [crystal-lang/crystal#5052]
`STABLE` or `UNSTABLE` flag keywords (just like `TODO`s) could be useful in docs.

**Comment by MakeNowJust** [comment]
Can you show the example?

**Comment by vladfaust** [comment]
@MakeNowJust according to https://github.com/crystal-lang/crystal/issues/5051#issuecomment-332681116, some methods could be marked `UNSTABLE`

**Comment by faustinoaq** [comment]
Yeah, so when all methods have "stable" flag we can prepare for 1.0 😅

**Comment by Sija** [comment]
@vladfaust According to what? There's nothing in that comment that would suggest it.
Also, `TODO` and others were meant to be used as a doc _flags_ (`FIXME: xxx`), not keywords.

**Comment by faustinoaq** [comment]
So, I have to rewrite my comment 😅 

when all `TODO` and `FIXME` flags are solved, we can prepare for 1.0

Currently there is:

- 50 `TODO:`
- 5 `FIXME:`
- 57 `NOTE:`

Crystal still have some great work to do for improving the language 💪

**Comment by vladfaust** [comment]
@Sija,
> Also, TODO and others were meant to be used as a doc flags (FIXME: xxx), not keywords.

From [documentation](https://crystal-lang.org/docs/conventions/documenting_code.html):
> The supported **flag keywords** are:

If a method is stable and we 100% know we wouldn't change it, we can add `STABLE` to it. Visa versa, using of `#future`, for example, is not recommended because it's likely to change, so we could mark it `UNSTABLE`.

---

@faustinoaq, in my opinion 1.0 means no any `UNSTABLE` flags if speaking of this issue :blush: 
Also `NOTE` doesn't mean something wrong, it's just a **note**.

**Comment by faustinoaq** [comment]
> Also NOTE doesn't mean something wrong, it's just a note.

Yeah, you're right! 😅

**Comment by vladfaust** [comment]
To sum it up, I believe that we need `UNSTABLE` keyword with optional description of why is it unstable. Developers will try to avoid unstable entities or keep in my that it might change with next release:

```crystal
# UNSTABLE: Windows support is likely to have strong changes prior to 1.0
class WindowsSupport
end

# Returns Fiber
# UNSTABLE: Experimental feature
def future
end
```

**Comment by straight-shoota** [comment]
Are these flags useful in the longrun? Or outside the compiler?

I don't really see much benefit from it. Things, that need to be done for 1.0 are not just APIs and this is tracked and managed outside the source code anyway, so why bother? And it's not very binding: Maybe a method was flagged `STABLE` but it turns out, it needs a refactoring for reasons not considered before.
